xo_install.sh

```diff
@@ -1,18 +1,18 @@
 #!/bin/bash
 
-#Repository add - confirm required repoistor(ies) are enabled and active, update and upgrade no matter what
-multiverse="http://archive.ubuntu.com/multiverse"
-if ! grep -q "^deb .*$multiverse"  /etc/apt/sources.list;
-then
-sudo add-apt-repository multiverse
-
 # Check if we were effectively run as root
 [ $EUID = 0 ] || { echo "This script needs to be run as root!"; exit 1; }
 
 #Check for 1GB Memory
 totalk=$(awk '/^MemTotal:/{print $2}' /proc/meminfo)
 if [ "$totalk" -lt "1000000" ]; then echo "XOCE Requires at least 1GB Memory!"; exit 1; fi 
 
+# Repository add - confirm required repoistor(ies) are enabled and active, update and upgrade no matter what
+multiverse="http://archive.ubuntu.com/multiverse"
```

Comment:
I think this line needs revised as grep is failing even if the repo already exists

Reply:
This will also fail for any Debian install.

Reply:
It worked on the 18.04.1-live Ubuntu I tested earlier, do you have any recommendations? As for Debian, adding the repository shouldn't be, completed, I agree but I'm not sure how to go about checking what distro we're on and then calling the specific function. 

I supposed we could check the /etc/*-release file with cat and pull a string from there to then pipe into an if statement.

Reply:
If by "worked" you mean "added the repo", then yes it worked. However, it will continue to do so each time you run that section of code because the grep command will always fail.

Reply:
Can you show me what you mean, as far as I can tell it works (on Ubuntu 18.04.1-live) still haven't progressed to other distro's for testing. 

![putty_2018-07-31_20-02-53](https://user-images.githubusercontent.com/14837699/43493611-d1dafbf4-94fc-11e8-8ef7-0216f458e395.png)

Reply:
The other alternative is we simply not bother with trying to fix this issue, and point people to the troubleshooting page with some predefined troubleshooting steps, such as checking to make sure that the correct repo's are available. . .

Reply:
I get the expected results. 

![virtualbox_2018-08-01_08-42-17](https://user-images.githubusercontent.com/14837699/43522259-d9a65d96-9566-11e8-9da6-ca6d5d092c0b.png)

Reply:
I don't see how that is possible when "http://archive.ubuntu.com/multiverse" doesn't appear in this file. 

I get similar results to you if I change the one line to this --
`multiverse="multiverse"`

Reply:
This line says If multiverse doesn't exist then do it. 

```if ! grep -q "^deb .*$multiverse"  /etc/apt/sources.list;```

Reply:
Then what's the purpose of this line?
`multiverse="http://archive.ubuntu.com/multiverse"`

I thought the bash variable (multiverse) was being "merged" in the grep command so that it was searching for the full value of $multiverse instead of the phrase "multiverse".

Reply:
There can be multiple lines that need multiverse, from the link you posted and one I found the variable adds it to multiple lines within the sources.list file. 

It works, but I kind of think it's a bad work around in general. 

Checking for dependencies I think would be the best approach, and stop if not found. On the other hand, single line install has always been a big feature of the installation script. (which really should only need to be run once)

Reply:
With that said of course. . . Ubuntu going and screwing the pooch on this and using a mini iso for their default download adds this complexity that didn't exist until 18.04.1-live on their website. 

The issue doesn't exist on other ISO's from what people were saying. I'd have to do more testing to confirm.

Reply:
That would be fine, but we also need to check if multiverse is included or not already, we don't want to add something to sources.list that already exists. 

And then the issue goes to what was an offline conversation do we simply want to check for minimum requirements and simply abort the install if the requirements aren't met.

Reply:
It doesn't actually add it again. If it already exists it just echoes "'multiverse' distribution component is already enabled for all sources."